doc/man3/OSSL_HPKE_CTX_new.pod

```diff
@@ -0,0 +1,525 @@
+     * handling
+     */
+    #define LBUFSIZE 48
+
+    /* Do a round-trip, generating a key, encrypting and decrypting */
+    int main(int argc, char **argv)
+    {
+        int ok = 0;
+        int hpke_mode = OSSL_HPKE_MODE_BASE;
+        OSSL_HPKE_SUITE hpke_suite = OSSL_HPKE_SUITE_DEFAULT;
+        OSSL_HPKE_CTX *sctx = NULL, *rctx = NULL;
+        EVP_PKEY *priv = NULL;
+        unsigned char pub[LBUFSIZE];
+        size_t publen = sizeof(pub);
+        unsigned char enc[LBUFSIZE];
+        size_t enclen = sizeof(enc);
+        unsigned char ct[LBUFSIZE];
+        size_t ctlen = sizeof(ct);
+        unsigned char clear[LBUFSIZE];
+        size_t clearlen = sizeof(clear);
+        const unsigned char *pt = "a message not in a bottle";
+        size_t ptlen = strlen((char *)pt);
+        const unsigned char *info = "Some info";
+        size_t infolen = strlen((char *)info);
+        unsigned char aad[] = { 1, 2, 3, 4, 5, 6, 7, 8 };
+        size_t aadlen = sizeof(aad);
+
+        /*
+         * Generate receiver's key pair.
+         * The receiver gives this public key to the sender.
+         */
+        if (OSSL_HPKE_keygen(NULL, NULL, hpke_suite, NULL, 0,
+                             pub, &publen, &priv) != 1)
+            goto err;
+
+        /* sender's actions - encrypt data using the receivers public key */
+        if ((sctx = OSSL_HPKE_CTX_new(hpke_mode, hpke_suite, NULL, NULL)) == NULL)
+            goto err;
+        if (OSSL_HPKE_encap(sctx, enc, &enclen, pub, publen, info, infolen) != 1)
+            goto err;
+        if (OSSL_HPKE_seal(sctx, ct, &ctlen, aad, aadlen, pt, ptlen) != 1)
+            goto err;
+
+        /* receiver's actions - decrypt data using the recievers private key */
+        if ((rctx = OSSL_HPKE_CTX_new(hpke_mode, hpke_suite, NULL, NULL)) == NULL)
+            goto err;
+        if (OSSL_HPKE_decap(rctx, enc, enclen, priv, info, infolen) != 1) 
+            goto err;
+        if (OSSL_HPKE_open(rctx, clear, &clearlen, aad, aadlen, ct, ctlen) != 1)
+            goto err;
+        ok = 1;
+    err:
+        /* clean up */
+        printf(ok ? "All Good!\n" : "Error!\n");
+        OSSL_HPKE_CTX_free(rctx);
+        OSSL_HPKE_CTX_free(sctx);
+        EVP_PKEY_free(priv);
+        return 0;
+    }
+
+=head1 WARNINGS
+
+Note that the OSSL_HPKE_CTX_set_seq() API could be dangerous - if used with GCM
+that could lead to nonce-reuse, which is a known danger. So avoid that
+entirely, or be very very careful when using that API.
+
+Use of an IKM value for deterministic key generation (via
+OSSL_HPKE_CTX_set1_ikme() or OSSL_HPKE_keygen()) creates the potential for
+leaking keys (or IKM values). Only use that if really needed and if you
+understand how keys or IKM values could be abused.
+
+=head1 SEE ALSO
+
+The RFC9180 specification: https://datatracker.ietf.org/doc/rfc9180/
+
+=head1 HISTORY
+
+Initially developed by Stephen Farrell supported by OTF via
```

Comment:
This should not be here I think.. Put it into the initial commit message.

It should say along the lines of..
This functionality described here was added in OpenSSL 3.2